@Ericson2314 did you tried this?

I still get the following error:
```
unpacking sources
patching sources
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/q3wdzrk9vran5mzzd9pcqyl43i1d90zw-aarch64-unknown-linux-android-ndk-gcc-binutils-wrapper-
patching script interpreter paths in /nix/store/q3wdzrk9vran5mzzd9pcqyl43i1d90zw-aarch64-unknown-linux-android-ndk-gcc-binutils-wrapper-
checking for references to /build/ in /nix/store/q3wdzrk9vran5mzzd9pcqyl43i1d90zw-aarch64-unknown-linux-android-ndk-gcc-binutils-wrapper-...
substituteStream(): WARNING: pattern '
' doesn't match anything in file '/nix/store/q3wdzrk9vran5mzzd9pcqyl43i1d90zw-aarch64-unknown-linux-android-ndk-gcc-binutils-wrapper-/nix-support/libc-ldflags'
building '/nix/store/i3hxsq3awwvam4pl95hc3iw4cn8dppr0-stdenv-linux.drv'...
building '/nix/store/ai01wnlh8g40phl3h378dil3x3z1y84a-hello-2.10-aarch64-unknown-linux-android.drv'...
unpacking sources
unpacking source archive /nix/store/3x7dwzq014bblazs7kq20p9hyzz0qh8g-hello-2.10.tar.gz
source root is hello-2.10
setting SOURCE_DATE_EPOCH to timestamp 1416139241 of file hello-2.10/ChangeLog
patching sources
updateAutotoolsGnuConfigScriptsPhase
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.guess
configuring
configure flags: --disable-dependency-tracking --prefix=/nix/store/s15p212cw9kjyy6m8klshd1rzcilnbc8-hello-2.10-aarch64-unknown-linux-android --build=x86_64-unknown-linux-gnu --host=aarch64-unknown-linux-android
checking for a BSD-compatible install... /nix/store/2f6zzvixj3wsj81flbdcpc6fdscgx4ai-coreutils-8.31/bin/install -c
checking whether build environment is sane... yes
checking for aarch64-unknown-linux-android-strip... aarch64-unknown-linux-android-strip
checking for a thread-safe mkdir -p... /nix/store/2f6zzvixj3wsj81flbdcpc6fdscgx4ai-coreutils-8.31/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for aarch64-unknown-linux-android-gcc... aarch64-unknown-linux-android-clang
checking whether the C compiler works... no
configure: error: in `/build/hello-2.10':
configure: error: C compiler cannot create executables
See `config.log' for more details
builder for '/nix/store/ai01wnlh8g40phl3h378dil3x3z1y84a-hello-2.10-aarch64-unknown-linux-android.drv' failed with exit code 77
cannot build derivation '/nix/store/dfy133728wcyf0ghmkiz356jjzv2ff8q-shell.drv': 1 dependencies couldn't be built
error: build of '/nix/store/dfy133728wcyf0ghmkiz356jjzv2ff8q-shell.drv' failed
```

Or is this something different?

--- a/lib/systems/examples.nix
+++ b/lib/systems/examples.nix
@@ -47,7 +47,7 @@ rec {
   armv7a-android-prebuilt = {
     config = "armv7a-unknown-linux-androideabi";
     sdkVer = "29";
-    ndkVer = "18b";
+    ndkVer = "21";
     platform = platforms.armv7a-android;
     useAndroidPrebuilt = true;
   };
@@ -55,7 +55,7 @@ rec {
   aarch64-android-prebuilt = {
     config = "aarch64-unknown-linux-android";
     sdkVer = "29";
-    ndkVer = "18b";
+    ndkVer = "21";
     platform = platforms.aarch64-multiplatform;
     useAndroidPrebuilt = true;
   };

--- a/pkgs/development/androidndk-pkgs/androidndk-pkgs.nix
+++ b/pkgs/development/androidndk-pkgs/androidndk-pkgs.nix
@@ -53,7 +53,9 @@ in
 
 rec {
   # Misc tools
-  binaries = runCommand "ndk-gcc-binutils" {
+  binaries = runCommand "ndk-toolchain-binutils" {
+    pname = "ndk-toolchain-binutils";
+    inherit (androidndk) version;
     isClang = true; # clang based cc, but bintools ld
     nativeBuildInputs = [ makeWrapper ];
     propagatedBuildInputs = [ androidndk ];

--- a/pkgs/development/mobile/androidenv/deploy-androidpackage.nix
+++ b/pkgs/development/mobile/androidenv/deploy-androidpackage.nix
@@ -5,7 +5,8 @@ let
   extraParams = removeAttrs args [ "package" "os" "buildInputs" "patchInstructions" ];
 in
 stdenv.mkDerivation ({
-  name = package.name + "-" + package.revision;
+  pname = package.name;
+  version = package.revision;
   src = if os != null && builtins.hasAttr os package.archives then package.archives.${os} else package.archives.all;
   buildInputs = [ unzip ] ++ buildInputs;
   preferLocalBuild = true;

